src/mlpack/methods/preprocess/preprocess_string_main.cpp

```diff
@@ -0,0 +1,155 @@
+
+PARAM_INT_IN("lowercase", "convert to lowercase(1 to convert).", "l", 1);
+PARAM_INT_IN("punctuation", "Remove punctuation (1 to remove).", "p", 1);
+PARAM_INT_IN("stopwords", "Remove stopwords (1 to remove).", "s", 1);
+PARAM_INT_IN("dimension", "Column which contains the string data.(1 by default)", "c", 1);
+
+using namespace mlpack;
+using namespace mlpack::util;
+using namespace arma;
+using namespace std;
+
+static void mlpackMain()
+{
+  // Parse command line options.
+
+  // Check on label parameters.
+
+  // Check for punctuation.
+  RequireParamValue<int>("punctuation",
+      [](double x) { return x == 0 || x <= 1; }, true,
+      "value for punctuation may be either 0 to not\
+       remove punctuation or 1 to remove punctuation 1");
+  RequireParamValue<int>("lowercase",
+      [](double x) { return x == 0 || x <= 1; }, true,
+      "value for lowercase may be either 0 to not\
+       remove punctuation or 1 to remove punctuation 1");
+  RequireParamValue<int>("stopwords",
+      [](double x) { return x == 0 || x <= 1; }, true,
+      "value for stopwords may be either 0 to not\
+       remove punctuation or 1 to remove punctuation 1");
+
+  // If puntucation is not set, warn the user.
+  if (!CLI::HasParam("punctuation"))
+  {
+    Log::Warn << "You did not specify " << PRINT_PARAM_STRING("punctuation")
+        << ", so it will be automatically set to 1." << endl;
+  }
+  // If lowercase is not set, warn the user.
+  if (!CLI::HasParam("lowercase"))
+  {
+    Log::Warn << "You did not specify " << PRINT_PARAM_STRING("lowercase")
+        << ", so it will be automatically set to 1." << endl;
+  }
+  // If stopwords is not set, warn the user.
+  if (!CLI::HasParam("stopwords"))
+  {
+    Log::Warn << "You did not specify " << PRINT_PARAM_STRING("stopwords")
+        << ", so it will be automatically set to 1." << endl;
+  }
+  // If dimension is not set, warn the user.
+  if (!CLI::HasParam("dimension"))
+  {
+    Log::Warn << "You did not specify " << PRINT_PARAM_STRING("dimension")
+        << ", so it will be automatically set to 1." << endl;
+  }
+  // Ectracting the filename
+  const std::string filename = CLI::GetParam<std::string>("actual_dataset");
+  // Extracting the Contents of file 
+      // File pointer 
+  fstream fin; 
+  
+  // Open an existing file 
+  fin.open(filename, ios::in); 
+  std::string line; 
+  std::vector<std::string> temp;
+  std::vector<std::string> input;
+  std::vector<std::vector<std::string>> dataset;
+  const int Column = CLI::GetParam<int>("dimension");
+  while (std::getline(fin, line))
+  { 
+    boost::split(temp, line, boost::is_any_of(std::string(1,',')));
+    input.push_back(temp[Column]);
+    dataset.push_back(temp);
+    temp.clear();
+  }
+  fin.close();
+  if(CLI::GetParam<int>("lowercase"))
+  {
+    data::LowerCase(input);
+  }
+  if(CLI::GetParam<int>("stopwords"))
+  {
+    // Not sure how to take input for tokenizer from cli.
+    //data::RemoveStopWords(input);
+  }
```

Comment:
Not sure, how should i build the binding for removing stop words, The binding takes a `.csv` file as input name and just parse and cleans the column having string data.

Reply:
Here we can first load stopwords and than can pass delimeter from data::Strtok class.
May be like this:
 ```
  if (CLI::HasParam("stopwords"))
  {
    std::unordered_set<std::string>stopword;
    stopword.insert("the");
    stopword.insert("a");    
    stopword.insert("isn't");
    stopword.insert("is");
    data::string_cleaning obj;
    obj.RemoveStopWords(input, stopword, data::Strtok(delimeter));
  }
```
Here I m loading stopwords intentionally. But you can load the data with the help some stopword file.
Thank you May this is helpful to you.

Reply:
> Here we can first load stopwords and than can pass delimeter from data::Strtok class.
> May be like this:
> 
> ```
>  if (CLI::HasParam("stopwords"))
>  {
>    std::unordered_set<std::string>stopword;
>    stopword.insert("the");
>    stopword.insert("a");    
>    stopword.insert("isn't");
>    stopword.insert("is");
>    data::string_cleaning obj;
>    obj.RemoveStopWords(input, stopword, data::Strtok(delimeter));
>  }
> ```
> 
> Here I m loading stopwords intentionally. But you can load the data with the help some stopword file.
> Thank you May this is helpful to you.

@Yashwants19 , the issues is not with passing of stopwords, the issue is - if you take a close look the function it implements custom tokenization, my question is how to take input for custom tokenization from cli or Python :)

Reply:
I would say it is okay to not have custom tokenization in the binding here.  Typically the bindings provide "simple" functionality that will cover most popular use cases, but for "complex" tasks the user might have to do it in C++, which is fine.

To @Yashwants19's point of stop words, I do think it would be useful to specify a file of stop words.  But the tricky thing is that this is a little awkward from Python, since right now to use this binding you would have to pass filenames---which is a bit weird since in Python one would expect the user to already have the stopwords as a loaded list of strings or something.

It's up to you if you want to handle this here, but one way around this would be to do the following: we can modify the documentation for `PARAM_VECTOR_STR_IN()` so that if the given vector has only one element and that element has `file:` as the first part, then the rest of it is a filename where we should load one string per line.  The way to do this would be to modify `src/mlpack/bindings/cli/get_param_impl.hpp` for the case where the type is `std::vector<std::string>`, and do the same for `src/mlpack/bindings/python/get_param_impl.hpp`.  In fact the function that checks if the vector contains only one element and then loads from file if needed can be put into some utility file in `src/mlpack/core/util/` or something and then just called from the `get_param_impl.hpp` file.  Anyway, I hope this makes sense; let me know if I can clarify the idea or anything.  And don't feel obligated to do it, but it could make the binding here cleaner to use.